app/src/main/java/com/github/se/eduverse/model/folder/FolderViewModel.kt

```diff
@@ -0,0 +1,95 @@
+package com.github.se.eduverse.model.folder
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+class FolderViewModel(val repository: FolderRepository) : ViewModel() {
+
+  private val _existingFolders: MutableStateFlow<MutableList<Folder>> =
+      MutableStateFlow(
+          repository
+              .getFolders(
```

Comment:
Instead of initializing _existingFolders with the result of repository.getFolders(), it's better to start with an empty list and update it asynchronously. This avoids potential issues with blocking the ViewModel initialization.

Reply:
Done